samples/DataImporter/GraphEngine.DataImporter/Importer.cs

```diff
@@ -216,33 +218,35 @@ private static void _Import<T>(IImporter<T> importer, string file, string typena
                  }
                  catch (Exception e)
                  {
-                     //Log.WriteLine(LogLevel.Error, "An error occured during import: \n{0}\n", e.Message);
+                     Log.WriteLine(LogLevel.Error, "An error occured during import: \n{0}\n", e.Message);
                  }
              });
         }
 
         public static IImporter GetImporter(string path)
         {
-            string filename = Path.GetFileName(path);
-            string filetype = Path.GetExtension(filename).ToLower();
-
-            if (filetype == ".gz" || filetype == ".zip")
+            IEnumerable<string> text = FileReader.ReadFile(path);
```

Comment:
We'd better make sure that the file is properly closed here. So I recommend explicitly calling GetEnumerator() and MoveNext() to acquire the first line, and then dispose the enumerator.